synapse/http/matrixfederationclient.py

```diff
@@ -924,7 +957,7 @@ async def post_json(
         timeout: Optional[int] = None,
         ignore_backoff: bool = False,
         args: Optional[QueryParams] = None,
-    ) -> Union[JsonDict, list]:
+    ) -> JsonDict:
```

Comment:
All of the callers want dicts anyway, so I figured we should just do it here instead of in each one.